Update on the metadata episode, first draft of introduction & setup

fpsom opened the pull request:
Metadata episode
- Added the "movie" exercise #9 
- Added bio.tool figure and info #15 #11 
- Added more definitions
- Added explanation of controlled vocabulary vs ontology #10 
- General fixes, typos and timing
- Addressed some of the issues in #51 

Introduction episode
- first draft

Setup page
- create accounts on GitHub and bio.tools

Guide page
- transferred points from the metadata episode

fpsom committed:
Update on the metadata episode, first draft of introduction & setup

- metadata: Added the "movie" exercise
- metadata: Added bio.tool figure and info
- metadata: Added more definitions
- metadata: Added explanation of controlled vocabulary vs ontology
- metadata: General fixes, typos and timing
- introduction: first draft
- setup: create accounts on GitHub and bio.tools
- guide: added only metadata points

fpsom committed:
Fixed minor typo.

mkuzak commented on _episodes/01-introduction.md:
```diff
@@ -1,11 +1,41 @@
 objectives:
-- "First objective."
+- "What are the basics of Open Science in research software"
+- "What are the FAIR principles"
```
Change objectives from questions to statements.

fpsom replied:
Comment addressed.

mkuzak commented on _episodes/01-introduction.md:
```diff
@@ -1,11 +1,41 @@
 exercises: 0
 questions:
-- "Key question"
+- "Why are best practices necessary in research software"
```
add "How Open Source can help with better quality of software?"

fpsom replied:
Comment addressed.

mkuzak commented on _episodes/03-use-registry.md:
```diff
@@ -26,33 +25,42 @@ You are already using metadata, but you might not be fully aware of it.
+- System documentation and
+- User documentation
+
+_System documentation_ represents documents that describe the system itself and its parts. It includes requirements documents, design decisions, architecture descriptions, program source code, and help guides. On the other hand, _User documentation_ covers manuals that are mainly prepared for end-users of the product and system administrators. User documentation includes tutorials, user guides, t ...
```
for me this documentation section is very formal

fpsom replied:
Comment addressed. Hopefully, the new text captures the same content but in a more simple format.

mkuzak commented on _extras/guide.md:
```diff
@@ -3,3 +3,10 @@ layout: page
+
+- Notes from the metadata episode
+  - [Local Installation of Zenodo](https://github.com/zenodo/zenodo/blob/master/INSTALL.rst)
+    It may be interesting to have a local installation of zenodo to play around. The instructions using Docker are available on the link above.
```
under the link

fpsom replied:
Comment addressed

mkuzak commented on _extras/guide.md:
```diff
@@ -3,3 +3,10 @@ layout: page
+    It may be interesting to have a local installation of zenodo to play around. The instructions using Docker are available on the link above.
+
+  - [Bio-Linux](http://environmentalomics.org/bio-linux-software-list/)
+    It is a final OS containing tools that have been already published, connected metadata, etc
```
what does "final OS" mean?

fpsom replied:
Comment addressed.

mkuzak commented on setup.md:
```diff
@@ -3,4 +3,15 @@ layout: page
 ---
-FIXME
+
+In order to be prepared for the lesson, you need to have accounts to the following (free) services:
```
accounts on

fpsom replied:
Comment addressed.

mkuzak commented on setup.md:
```diff
@@ -3,4 +3,15 @@ layout: page
+
+2. BioTools
+
+[bio.tools](https://bio.tools/) is a portal to bioinformatics resources worldwide, aimed to help bioinformaticians and scientists, find, understand, compare and select resources as well as use and connect them in workflows.
```
they don't need bio.tools account only dev.bio.tools

fpsom replied:
We are actually explicitly requesting for participants to create an account on the dev instance in the next sentence. However, I thought it made sense to include an intro to the actual platform. If that is redundant, we can delete it.

fpsom committed:
Fixed points raised by @mkuzak

fpsom commented:
Thanks for the review @mkuzak ! I think I've addressed all the issues (with the possible exception of the bio.tools one, but we can revisit this for sure).

ljgarcia commented on _episodes/01-introduction.md:
```diff
@@ -1,11 +1,42 @@
 objectives:
-- "First objective."
+- "Basics of Open Science in research software"
+- "Introduction to the FAIR principles"
```
FAIR principles are related to metadata, I suggest to add a key question about the role played by metadata in research software

fpsom replied:
Given that this introduction is for all four of the best practices, I think that it may be too specific to have a question just for metadata. Thoughts?

ljgarcia replied:
Totally agree, you are right, no need to add anything here

ljgarcia commented on _episodes/01-introduction.md:
```diff
@@ -1,11 +1,42 @@
 keypoints:
-- "First key point."
+- "Best practices in research software are tied to the FAIR principles"
+- "They are not tailored to software developers, but rather to a wider audience"
```
Not sure whether "they" here refers to the best practices or the FAIR principles.

fpsom replied:
I've addressed this with a new commit.

ljgarcia commented on _episodes/01-introduction.md:
```diff
@@ -1,11 +1,42 @@
+- "They are not tailored to software developers, but rather to a wider audience"
 ---
+
+Scientific research relies on computer software, yet software is not always developed following practices that ensure its quality and sustainability. One of the most recent publications ([Four simple recommendations to encourage best practices in research software](https://f1000research.com/articles/6-876/v1)) provided a simple, yet robust framework of simple recommendations that encourage the ad ...
```
"Scientific research relies on computer software" Do you mean it relies on regular software development practices?

fpsom replied:
Rephrased to better reflect the original intention.

ljgarcia commented on _episodes/01-introduction.md:
```diff
@@ -1,11 +1,42 @@
+- "They are not tailored to software developers, but rather to a wider audience"
 ---
+
+Scientific research relies on computer software, yet software is not always developed following practices that ensure its quality and sustainability. One of the most recent publications ([Four simple recommendations to encourage best practices in research software](https://f1000research.com/articles/6-876/v1)) provided a simple, yet robust framework of simple recommendations that encourage the ad ...
```
"One of the most recent publications" I would suggest to omit "most".

fpsom replied:
I've addressed this with a new commit.

ljgarcia commented on _episodes/01-introduction.md:
```diff
@@ -1,11 +1,42 @@
+
+Scientific research relies on computer software, yet software is not always developed following practices that ensure its quality and sustainability. One of the most recent publications ([Four simple recommendations to encourage best practices in research software](https://f1000research.com/articles/6-876/v1)) provided a simple, yet robust framework of simple recommendations that encourage the ad ...
+
+Based on these recommendations, this lesson focuses on providing both the underlying context as well as some practical exercises towards establishing their usefulness in the long term. The consequent episodes of this lesson are structured in the form of one episode per recommendation;
```
Colon rather than semi-colon at the end of this paragraph?

fpsom replied:
Fixed in the new commit.

ljgarcia commented on _episodes/01-introduction.md:
```diff
@@ -1,11 +1,42 @@
+
+"_When all researchers are aware of Open Science, and are trained, supported and guided at all career stages to practice Open Science, the potential is there to fundamentally change the way research is performed and disseminated, fostering a scientific ecosystem in which research gains increased visibility, is shared more efficiently, and is performed with enhanced research integrity._" [Open Sci ...
+
+Discussing best practices in developing research software, one is bound to touch on the subject of Open Science. Modern research relies on software, and building upon—or reproducing—that research requires access to the full source code behind that software ([ref](https://open-science-training-handbook.gitbook.io)). Sharing software used for research (whether computational in nature, or that relie ...
```
upon-or? Not sure if should be upon -or (long dash there)

fpsom replied:
Fixed in the new commit

ljgarcia commented on _episodes/01-introduction.md:
```diff
@@ -1,11 +1,42 @@
+
+Discussing best practices in developing research software, one is bound to touch on the subject of Open Science. Modern research relies on software, and building upon—or reproducing—that research requires access to the full source code behind that software ([ref](https://open-science-training-handbook.gitbook.io)). Sharing software used for research (whether computational in nature, or that relie ...
+
+## FAIR principles
```
I suggest to make explicit the relation between the 4OSS and FAIR, otherwise is kind of disconnected

fpsom replied:
Fixed in the new commit.

ljgarcia commented on _episodes/03-use-registry.md:
```diff
@@ -78,25 +101,21 @@ A standard can be defined as "a structure agreed and adopted by a community" or
 
-TODO: difference between control vocabulary and ontology.
-
+**Controlled vocabularies** provide a way to organize knowledge for subsequent retrieval. It is usually a carefully selected list of words and phrases, which are used to tag units of information (document or work) so that they may be more easily retrieved by a search. The fundamental difference between an **ontology** and a **controlled vocabulary** is the level of abstraction and relationships a ...
```
If we are going to explicitlely mention ontologies, we might mention as well others, explaining how they go from soft/weak representations to those with stronger logic in it, and then move to ontologies and their importance.

fpsom replied:
I left this unresolved as I think it may be too technical. Thoughts?

ljgarcia replied:
It would be indeed too technical. What about "The fundamental difference between and **ontology** and other **controlled vocabularies**, e.g., thesauri, is the [...]"?

fpsom replied:
That is a great point! Fixed in a new commit.

fpsom committed:
Merge branch 'gh-pages' into fpsom-general-patch

fpsom commented:
Updated PR to reflect @tobyhodges comment in #44.

ljgarcia commented on _episodes/03-use-registry.md:
```diff
@@ -197,17 +214,9 @@ Zenodo is a general-purpose open access repository.
-
-- [Bio-Linux](http://environmentalomics.org/bio-linux-software-list/)
-It is a final OS containing tools that have been already published, connected metadata, etc
+By adding good enough metadata to our research software, we are directly supporting its findability, thus increasing the overall visibility of the software. This is tied to the **findable** aspect of the FAIR principles mentioned in the introductory episode of this lesson.
```
Metadate can also support accessibility if you include a license there, or interoperability if you include input/output data types or format. There might be some metadata supporting as well reusability.

That would be a nice exercise, asking attendees to map the metadata they have identified to FAIR principles... but maybe not enough time for it though...

fpsom replied:
Fixed in the new commit. And excellent idea for an optional challenge @ljgarcia ! :)

fpsom committed:
Addressed some of the comments raised by @ljgarcia

fpsom committed:
Additional fixes proposed by @ljgarcia

fpsom commented:
Thanks for the comments @ljgarcia ! I've addressed most of them in a new commit, with the exception of two.

fpsom committed:
Additional fix addressing @ljgarcia comment

fpsom commented:
Thanks @tobyhodges , @ljgarcia and @mkuzak for the comments and the review. If you are happy with the current version, feel free to merge.

fpsom commented:
I'll take the silence as acceptance, so I'll merge the PR by the end of day today. :)

FatmaZG commented:
Thanks to all for the great improvements I've seen to the first draft. @fpsom we might still have to shrink down a bit the list of examples (great new ones added by @orchid00). Any final timing for each section/sub-section?
